cachi2/core/models/sbom.py

```diff
@@ -87,20 +86,624 @@ class Metadata(pydantic.BaseModel):
     tools: list[Tool] = [Tool(vendor="red hat", name="cachi2")]
 
 
+def spdx_now() -> str:
+    """Return a time stamp in SPDX-compliant format."""
+    return datetime.datetime.now().isoformat()[:-7] + "Z"
+
+
 class Sbom(pydantic.BaseModel):
     """Software bill of materials in the CycloneDX format.
 
     See full specification at:
     https://cyclonedx.org/docs/1.4/json
     """
 
+    model_config = pydantic.ConfigDict(extra="forbid")
+
     bom_format: Literal["CycloneDX"] = pydantic.Field(alias="bomFormat", default="CycloneDX")
     components: list[Component] = []
     metadata: Metadata = Metadata()
     spec_version: str = pydantic.Field(alias="specVersion", default="1.4")
     version: int = 1
 
+    def __add__(self, other: Union["Sbom", "SPDXSbom"]) -> "Sbom":
+        if isinstance(other, self.__class__):
+            return Sbom(
+                components=merge_component_properties(
+                    chain.from_iterable(s.components for s in [self, other])
+                )
+            )
+        else:
+            return self + other.to_cyclonedx()
+
     @pydantic.field_validator("components")
     def _unique_components(cls, components: list[Component]) -> list[Component]:
         """Sort and de-duplicate components."""
         return unique_sorted(components, by=lambda component: component.key())
+
+    def to_cyclonedx(self) -> Self:
+        """Return self, self is already the right type of Sbom."""
+        # This is a short-cut, but since it is unlikely that we would ever add more Sbom types
+        # it is acceptable. If, however this ever happens a proper base class will be needed.
+        return self
+
+    def to_spdx(self, doc_namespace: str) -> "SPDXSbom":
+        """Convert a CycloneDX SBOM to an SPDX SBOM.
+
+        Args:
+            doc_namespace: SPDX document namespace. Namespace is URI of indicating
+
+        """
+
+        def create_document_root() -> SPDXPackage:
+            return SPDXPackage(name="", versionInfo="", SPDXID="SPDXRef-DocumentRoot-File-")
```

Comment:
Is this element required by the spec? I tried to find references to it, but couldn't. I was also wondering why does it end with a hyphen?

Reply:
I think I found the [reference](https://github.com/midnightercz/konflux-ci-architecture/blob/571c093b701c0467d60c75cf0f2f7355a17c49df/ADR/0044-spdx-support.md?plain=1#L150-L151) for it in the Konflux SPDX design (which triggered this change here). 

So the intention was to inform the folder that was processed to generate the SBOM. If so, I think we want to keep it as "SPDXRef-DocumentRoot-**Directory**-{dirname}", and maybe use the same `dirname` for the "name" package attribute.

Reply:
It might make more sense to have one "main element" for each "main package". So `cachi2 fetch-deps '[{"type": "pip"}, {"type": "npm"}]'` would result in something like the example here https://github.com/konflux-ci/build-tasks-dockerfiles/tree/main/sbom-utility-scripts/scripts/add-image-reference-script#list-of-updates

Reply:
But that would need more changes I think, so a made-up main element is fine in the meantime.

But if you do use `dirname` for the name attribute, please use `./{dirname}` to clearly make it look like a file rather than an actual package

Reply:
> It might make more sense to have one "main element" for each "main package

I like this idea, it could be a future improvement.